.github/workflows/formatter.yml

```
@@ -0,0 +1,43 @@
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Apply Spotless formatting
+        run: mvn spotless:apply
+
+      - name: Commit changes if any
+        run: |
```

actionlint (1.7.8)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)